### ISSUE TITLE
Test_center reruns all XCUITests instead of just the failed ones, and doesn't report the build as failed

### ISSUE THREAD

**Issue by scotthmccoy** [lyndsey-ferguson/fastlane-plugin-test_center#99]
### Question Checklist

- [✅] Updated `fastlane-plugin-test_center` to the latest version
- [✅] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [✅] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [✅] I searched for [existing GitHub issues](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/issues)

### Question Subject

Environment:
- I'm testing the retry feature by including an `XCTFail()` in `QA_TestCases_01_04_67_basicBanner.testGo`. 
- The test `QA_TestCases_03_BackgroundingApp.testGo()` is also pretty flaky. 
- The `try_count` param is set to 3 (I'm using the Fastfile from the example with my project paths subbed in).
- My full suite of XCUITests take about 500 seconds (9 minutes) to run.

Desired Behavior:
- With `try_count` set to 3, all UI tests should be run once, then only the tests that failed are re-run on the next two runs. A build should be failed only if the same test fails all 3 times.

Observed Behavior:
- In my [Jenkins Log](https://drive.google.com/file/d/14WkK6o7vf_v9wsqoA1INbpfCyLI1SnvO/view?usp=sharing), at 10:41:33 I see xcodebuild invoked with a bunch of `-only-testing` args (good)
- At 10:49:58 and 10:58:47 I see xcodebuild invoked again, but with *no* `-only-testing` args. It runs all the tests based on the time it takes to complete the test run (about 9 minutes each time).
- I'm also seeing `Number of tests    | 0 ` after each test run, which is odd.

Here is my Fastfile:
```
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

lane :sweep do
  
  test_run_block = lambda do |testrun_info|
    failed_test_count = testrun_info[:failed].size
    
    if failed_test_count > 0
      UI.important('The run of tests would finish with failures due to fragile tests here.')

      try_attempt = testrun_info[:try_count]
      if try_attempt < 3
        UI.header('Since we are using :multi_scan, we can re-run just those failing tests!')
      end
    end
  end
  
  result = multi_scan(
    workspace:"VrtcalSDK.xcworkspace",
    scheme: "VrtcalSDKInternalTestApp",
    try_count: 3,
    fail_build: false,
    testrun_completed_block: test_run_block
  )
  
  unless result[:failed_testcount].zero?
    UI.message("There are #{result[:failed_testcount]} legitimate failing tests")
  end
end
```

**Comment by scotthmccoy** [comment]
I was able to get the build to report as failed by turning on `fail_build:true`. I feel a little sheepish about that. -_-

**Comment by lyndsey-ferguson** [comment]
1. can you turn off color, the logs hurt my eyes with all the color codes :). If you set the ENVIRONMENT variable `FASTLANE_DISABLE_COLORS=1` that should do it.
2. Even though you seem to be using the latest version, `multi_scan` seems to have a problem with your `xctestrun` file as I see this: `There are no tests to run in testable '__xctestrun_metadata__'`. That used to happen when Apple introduced a new entry in the file and the plugin didn't know about it. Would you be able to attach that file (sensitive data redacted)?
3. As you saw, the `--only-testing` is missing. The summary also indicates that it is empty: `only_testing                                   | []`.  Can you, in your `test_run_block` add code to print the contents of the junit files? `multi_scan` looks for failing tests in the junit file, and if there are none, well, it cannot provide a set of tests to run.

**Comment by lyndsey-ferguson** [comment]
Oh, and if you're trying to use this with simulators running in parallel, I'm not finished that feature yet (and it is not part of the 3.6.3 release).

**Comment by lyndsey-ferguson** [comment]
You _could_ try playing with it by reading this: https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/issues/10

**Comment by scotthmccoy** [comment]
> Oh, and if you're trying to use this with simulators running in parallel, I'm not finished that feature yet (and it is not part of the 3.6.3 release).

I've got 3 clones running in parallel. Would that result in the behavior I'm seeing?

**Comment by lyndsey-ferguson** [comment]
Yes, there is a problem with `scan` and Xcode Schemes that have been configured to run testsuites in parallel: the junit reports are not produced and it is very difficult to get the failed tests. There probably is another way via test summaries.plist, but that is more work.

**Comment by lyndsey-ferguson** [comment]
I think that I will close this Issue and create a new Feature Request to abort if it is detected if the Xcode Scheme has requested parallel builds.

**Comment by scotthmccoy** [comment]
That's strange, it seems to be working well with the exception of re-running all the tests instead of just the failed ones. Any idea when multi-sim runs with test reruns will be fully supported? I'd dearly love to have that!

**Comment by lyndsey-ferguson** [comment]
The problem is that `xcodebuild` when managing its own parallel execution of testing, does not output in a way that the `xcpretty` gem can parse and create junit files from.

I am working on multi_scan's parallelization feature as we speak. Given that I spend a small amount of time (consistently) on it, I cannot really say. What I can say is that I do have it functional, but it is not complete.

Visit the issue to try it if you'd like.

**Comment by lyndsey-ferguson** [comment]
Created this ticket: [Feature Request: abort multi_scan if the Xcode Scheme specifies parallelization](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/issues/101)
